### PR TITLE
Replace MD5 with MurmurHash3_128 in SPMI

### PR DESCRIPTION
**Pull request by EgorBo**
This PR removes MD5 usages from SPMI and replaces those with a simple MurMur3 hash implementation copied from https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash3.cpp#L255 with a few changes:

1) `getblock64` is implemented via `memcpy` to avoid UB around strict-aliasing/unaligned reads (thanks @jakobbotsch for noticing that)
2) Same for [these stores](https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash3.cpp#L330-L331).
3) All unused functions are removed
4) removed ifdefs around `rotl64 `, MSVC can recognize that idiom from code itself: https://godbolt.org/z/zj5E55sW3

NOTE: it's not cryptography/security related, we just needed a simple 128bit hash with a good distribution for an internal tool we use for jit testing.

### CODE REVIEW

**Comment by EgorBo** [comment]
PTAL @BruceForstall @jakobbotsch
cc @GrabYourPitchforks

File: THIRD-PARTY-NOTICES.TXT [hunk 1]
```diff
@@ -1206,3 +1206,29 @@ As an exception, if, as a result of your compiling your source code, portions
 of this Software are embedded into a machine-executable object form of such
 source code, you may redistribute such embedded portions in such object form
 without including the above copyright and permission notices.
+
+
+License for MurmurHash3 (https://github.com/aappleby/smhasher)
```
Comment (jkotas) [inline THIRD-PARTY-NOTICES.TXT:1211]: We do not need the whole smhasher test suite. We can just reference the specific file that we need and that is under a different license:

https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp#L2-L3

File: src/coreclr/tools/superpmi/superpmi-shared/hash.cpp [hunk 2]
```diff
@@ -2,87 +2,110 @@
...
+    k *= 0xff51afd7ed558ccdLLU;
+    k ^= k >> 33;
+    k *= 0xc4ceb9fe1a85ec53LLU;
+    k ^= k >> 33;
+    return k;
 }
 
-// static
-bool Hash::Destroy()
+void MurmurHash3_128(const void* key, const size_t len, const uint32_t seed, void* out)
```
Comment (jakobbotsch) [inline src/coreclr/tools/superpmi/superpmi-shared/hash.cpp:47]:
```suggestion
static void MurmurHash3_128(const void* key, const size_t len, const uint32_t seed, void* out)
```

File: THIRD-PARTY-NOTICES.TXT (hunk 1 above)
Comment (EgorBo) [reply THIRD-PARTY-NOTICES.TXT]: Thanks, updated, let me know if it needs better wording

File: THIRD-PARTY-NOTICES.TXT [hunk 3]
```diff
@@ -1206,3 +1206,11 @@ As an exception, if, as a result of your compiling your source code, portions
...
 source code, you may redistribute such embedded portions in such object form
 without including the above copyright and permission notices.
+
+
+License for MurmurHash3
+--------------------------------------
+
+MurmurHash3 was written by Austin Appleby, and is placed in the public
+domain. The author hereby disclaims copyright to this source code: 
+https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash3.cpp
```
Comment (jkotas) [inline THIRD-PARTY-NOTICES.TXT:1216]:
```suggestion
https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp

MurmurHash3 was written by Austin Appleby, and is placed in the public
domain. The author hereby disclaims copyright to this source ```

This matches the format used in other entries.

**Review by jkotas — APPROVED** [review]
LGTM!

**Comment by BruceForstall** [comment]
I think this should have changed the JIT-EE GUID since I think the hash is embedded in the TOC (.mct) files among other things. But the GUID has changed several times since then, so nothing to do now.

**Pull request merged**